src/NexusMods.App.UI/Pages/Downloads/DownloadsPageViewModel.cs

```diff
@@ -11,28 +18,206 @@ namespace NexusMods.App.UI.Pages.Downloads;
+            // NOTE(sewer): This is a hack.
+            //
+            // When we pause/resume a selected item, there's a chance that GetDownloadsByStatus
+            // will fire before the status of the actual job changes. This means that the evaluation
+            // result will not be correct, so we need to delay for a short moment to make sure it is.
+            // 
+            // As far as workarounds go, we can't subscribe to DownloadComponents.StatusComponent directly because
+            // the JobStatus field there is derived from the model, i.e. also from GetDownloadsByStatus(JobStatus.Running)
+            //
+            // We can't guarantee a specific execution order; so we wait with a small hack.
+            .Delay(TimeSpan.FromMilliseconds(64))
+            .OnUI()
+            .Select(_ => Unit.Default);
+        
+        // Create observables that track selection and status changes
+        var selectionStatusReevaluationTrigger = this.WhenAnyValue(vm => vm.SelectionCount)
+            .Select(_ => Unit.Default)
+            .Merge(runningCountChanges);
+
+        SelectionHasRunningItems = selectionStatusReevaluationTrigger
+            .Select(_ => Adapter.SelectedModels
+                .Select(model => model.GetOptional<DownloadComponents.StatusComponent>(DownloadColumns.Status.ComponentKey))
+                .Where(opt => opt.HasValue)
+                .Any(opt => opt.Value.Status.Value == JobStatus.Running))
+            .ToObservable();
+        
+        SelectionHasPausedItems = selectionStatusReevaluationTrigger
+            .Select(_ => Adapter.SelectedModels
+                .Select(model => model.GetOptional<DownloadComponents.StatusComponent>(DownloadColumns.Status.ComponentKey))
+                .Where(opt => opt.HasValue)
+                .Any(opt => opt.Value.Status.Value == JobStatus.Paused))
+            .ToObservable();
+        
+        SelectionHasActiveItems = selectionStatusReevaluationTrigger
+            .Select(_ => Adapter.SelectedModels
+                .Select(model => model.GetOptional<DownloadComponents.StatusComponent>(DownloadColumns.Status.ComponentKey))
+                .Where(opt => opt.HasValue)
+                .Any(opt => opt.Value.Status.Value.IsActive()))
+            .ToObservable();
+        
+        HasRunningItems = downloadsService.GetDownloadsByStatus(JobStatus.Running)
+            .QueryWhenChanged(items => items.Count > 0)
+            .OnUI()
+            .ToObservable()
+            .Prepend(false);
+        
+        HasPausedItems = downloadsService.GetDownloadsByStatus(JobStatus.Paused)
+            .QueryWhenChanged(items => items.Count > 0)
+            .OnUI()
+            .ToObservable()
+            .Prepend(false);
+
+        // Commands with CanExecute
+        PauseAllCommand = HasRunningItems.ToReactiveCommand<Unit>(_ => downloadsService.PauseAll());
+        ResumeAllCommand = HasPausedItems.ToReactiveCommand<Unit>(_ => downloadsService.ResumeAll());
```

Comment:
Since this page can be scoped by game, should these only pause/resume items for that game?

I'm thinking, if someone wants to pause all downloads from a game to focus on another game.